include/LightGBM/boosting.h

```diff
@@ -312,6 +312,8 @@ class LIGHTGBM_EXPORT Boosting {
   * \return The boosting object
   */
   static Boosting* CreateBoosting(const std::string& type, const char* filename);
+
+  virtual bool IsLinear() const { return false; }
```

Comment:
It's used in the `refit` method of the python interface. When we construct the new dataset for refitting, we need to know if the model is linear, so that we can save the full feature data not just the binned data.